ste/xferRetrypolicy.go

```diff
@@ -392,6 +395,12 @@ func NewBlobXferRetryPolicyFactory(o XferRetryOptions) pipeline.Factory {
 				switch {
 				case err == nil:
 					action = "NoRetry: successful HTTP request" // no error
+				case response.Response().StatusCode == http.StatusBadRequest:
+					if o.RetryBadRequestCustom && os.Getenv("AZCOPY_DISABLE_TRANSFER_RETRY_BAD_REQUEST") == "" {
```

Comment:
Maybe cache the result of os.Getenv("AZCOPY_DISABLE_TRANSFER_RETRY_BAD_REQUEST")?